docs/source/_src_user_guide/state_propagation/propagation_setup/acceleration_models/thrust.rst

```diff
@@ -0,0 +1,301 @@
+
+.. warning:: The direction vector that is being returned by the custom function should be a unit vector.
+
+Custom thrust orientation settings
+==================================
+
+Thrust orientation settings can also be created trough a custom function that returns this time not the direction but the orientation of the thrust.
+
+This thrust orientation needs to be provided through a rotation matrix representing the rotation from body-fixed thrust direction to the inertial thrust direction.
+
+   .. tabs::
+
+      .. tab:: Python
+
+         The Tudat(Py) API docs give more details on the :func:`~tudatpy.numerical_simulation.propagation_setup.thrust.custom_thrust_orientation` function.
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/custom_thrust_orientation.py
+            :language: python
+
+      .. tab:: C++
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/custom_thrust_orientation.cpp
+            :language: cp
+
+Thrust direction from existing orientation
+==========================================
+
+The orientation of the vehicle is in some cases already defined. This could be thanks to aerodynamic guidance or to the propagation of rotational dynamics. In this context, the thrust direction can be computed from the body-fixed direction. 
+
+In such a case, the thrust direction is computed from the existing vehicle orientation.
+Do note that an additional angle from the vehicle can be defined, for instance in case Thrust Vectoring Control is used.
+This angle, the body fixed thrust direction, can be defined in the :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustMagnitudeSettings` class.
+
+This thrust direction does not require a specific derived class, but instead only requires the use of the following function:
+
+   .. tabs::
+
+      .. tab:: Python
+
+         The Tudat(Py) API docs give more details on the :func:`~tudatpy.numerical_simulation.propagation_setup.thrust.thrust_from_existing_body_orientation` function.
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/from_existing_orientation.py
+            :language: python
+
+      .. tab:: C++
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/from_existing_orientation.cpp
+            :language: cp
+
+Thrust magnitude
+~~~~~~~~~~~~~~~~
+
+Below, the different methods that have been implemented to define the magniude of the thrust are outlined. Note that these settings are *only* relevant if you use the :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_direction_and_magnitude` function.
+	
```

Comment:
I'd specify here that all these methods return a `ThrustMagnitudeSettings` object.